client/src/templates/Challenges/quiz/show.tsx

```diff
@@ -111,147 +165,148 @@ class ShowQuiz extends Component<ShowQuizProps, ShowQuizState> {
       helpCategory
     });
     challengeMounted(challengeMeta.id);
-    this.container.current?.focus();
-  }
+    container.current?.focus();
+    // This effect should be run once on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  componentDidUpdate(prevProps: ShowQuizProps): void {
-    const {
-      data: {
-        challengeNode: {
-          challenge: { title: prevTitle }
-        }
-      }
-    } = prevProps;
-    const {
-      challengeMounted,
-      data: {
-        challengeNode: {
-          challenge: { title: currentTitle, challengeType, helpCategory }
+  useEffect(() => {
+    updateChallengeMeta({
+      ...challengeMeta,
+      title,
+      challengeType,
+      helpCategory
+    });
+    challengeMounted(challengeMeta.id);
+  }, [
+    title,
+    challengeMeta,
+    challengeType,
+    helpCategory,
+    challengeMounted,
+    updateChallengeMeta
+  ]);
+
+  const handleAnswersCheck = () => {
+    const unansweredQuestions = quizData.reduce(
+      (accumulator, currentQuestion, currentIndex) => {
+        if (currentQuestion.selectedAnswer == null) {
+          accumulator.push(currentIndex + 1);
         }
+        return accumulator;
       },
-      pageContext: { challengeMeta },
-      updateChallengeMeta
-    } = this.props;
-    if (prevTitle !== currentTitle) {
-      updateChallengeMeta({
-        ...challengeMeta,
-        title: currentTitle,
-        challengeType,
-        helpCategory
-      });
-      challengeMounted(challengeMeta.id);
+      [] as number[]
```

Comment:
```suggestion
      []
```
Another nit. No need for an assertion if you provide the type to reduce.